src/Castle.Windsor/MicroKernel/Context/CreationContext.cs

```diff
@@ -67,7 +67,7 @@ public class CreationContext :
 		/// <param name = "parentContext"> The parent context. </param>
 		/// <param name = "propagateInlineDependencies"> When set to <c>true</c> will clone <paramref name = "parentContext" /> <see cref = "AdditionalArguments" /> . </param>
 		public CreationContext(Type requestedType, CreationContext parentContext, bool propagateInlineDependencies)
-			: this(parentContext.Handler, parentContext.ReleasePolicy, requestedType, null, null, parentContext)
+			: this(parentContext?.Handler, parentContext?.ReleasePolicy, requestedType, null, null, parentContext)
```

Comment:
The problem with doing this is that you call the other constructor passing a null `IHandler` and `IReleasePolicy` which isn't valid.

You'll find that Windsor likely has a bunch of these "bugs" that aren't hurting anyone because Windsor's original design was to be completely open with everything marked public, you can see `ResolutionContext` in the same file is public with no checking, but is never created by users. We've been slowly closing things with Castle DynamicProxy because we needed to to keep the public API sane and make changes, however Windsor is still completely open.

I'd like to see Windsor enable nullable reference types, however accepting this pull request would mean that the other constructor needs to accept null and run to completion to then run this constructor which throws.